### ISSUE TITLE
Can add multiple ssh keys with the same name and remove throws an exception

### ISSUE THREAD

**Issue by LoicMahieu** [deis/deis#4408]
````
--- VERSIONS ---
CoreOS 723.3.0
Deis v1.9.1
etcd version 0.4.9
fleetd version 0.10.2
```

---

With travis/dpl I have been able to add different ssh keys with the same name:
```
./deis keys
=== igloobot Keys
1c5301e127f6 ssh-rsa AAAAB3Nz...1Oz6gVwvEB
1c5301e127f6 ssh-rsa AAAAB3Nz...3MMq4g9c/n
1c5301e127f6 ssh-rsa AAAAB3Nz...+6pLYwHPMZ
```

And the controller throw a 500 when I try to delete... one of them? :
```
./deis keys:remove 1c5301e127f6
Removing 1c5301e127f6 SSH Key... 500 INTERNAL SERVER ERROR
```

Controller logs:
```
Sep 04 13:53:48 deis-1 sh[6209]: ERROR Internal Server Error: /v1/keys/1c5301e127f6
Sep 04 13:53:48 deis-1 sh[6209]: Traceback (most recent call last):
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/django/core/handlers/base.py", line 112, in get_response
Sep 04 13:53:48 deis-1 sh[6209]: response = wrapped_callback(request, *callback_args, **callback_kwargs)
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 57, in wrapped_view
Sep 04 13:53:48 deis-1 sh[6209]: return view_func(*args, **kwargs)
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/rest_framework/viewsets.py", line 85, in view
Sep 04 13:53:48 deis-1 sh[6209]: return self.dispatch(request, *args, **kwargs)
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/rest_framework/views.py", line 407, in dispatch
Sep 04 13:53:48 deis-1 sh[6209]: response = self.handle_exception(exc)
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/rest_framework/views.py", line 404, in dispatch
Sep 04 13:53:48 deis-1 sh[6209]: response = handler(request, *args, **kwargs)
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/rest_framework/mixins.py", line 84, in destroy
Sep 04 13:53:48 deis-1 sh[6209]: instance = self.get_object()
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/rest_framework/generics.py", line 242, in get_object
Sep 04 13:53:48 deis-1 sh[6209]: obj = get_object_or_404(queryset, **filter_kwargs)
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/rest_framework/generics.py", line 34, in get_object_or_404
Sep 04 13:53:48 deis-1 sh[6209]: return _get_object_or_404(queryset, *filter_args, **filter_kwargs)
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/django/shortcuts/__init__.py", line 113, in get_object_or_404
Sep 04 13:53:48 deis-1 sh[6209]: return queryset.get(*args, **kwargs)
Sep 04 13:53:48 deis-1 sh[6209]: File "/usr/lib/python2.7/site-packages/django/db/models/query.py", line 313, in get
Sep 04 13:53:48 deis-1 sh[6209]: (self.model._meta.object_name, num))
Sep 04 13:53:48 deis-1 sh[6209]: MultipleObjectsReturned: get() returned more than one Key -- it returned 3!
Sep 04 13:53:48 deis-1 sh[6209]: 10.244.66.1 "DELETE /v1/keys/1c5301e127f6 HTTP/1.0" 500 27 "python-requests/2.5.1 CPython/2.7.6 Linux/4.0.9-boot2docker"
```

---

Quick fix ?
- Can I remove the key manually in the database ?

**Comment by LoicMahieu** [comment]
I would like to fix the situation and keep going on my deis deployments.
How can I remove these keys ? Is it enough to delete records in database ?

**Comment by aj-may** [comment]
@LoicMahieu I had the same issue.  I was able to remove the duplicates by `nse`-ing into the deis-controller container and deleting them through the Django shell. `./manage.py shell`

**Comment by LoicMahieu** [comment]
Thanks @aj-may I have a direct access to the database so if I remove the keys with that way it would be the same that with the Django shell. Thanks!

**Comment by aj-may** [comment]
@LoicMahieu had to check the model.  Looks like removing the rows from the DB should work just the same.

**Comment by bacongobbler** [comment]
It's best to use the shell. The controller calls some post-dekete code afterwards to remove the key from etcd.

**Comment by aj-may** [comment]
@bacongobbler ahh.. forgot to check for signals

**Comment by LoicMahieu** [comment]
Ok thanks. For users that didn't know django shell (like me):
```sh
$ python manage.py shell
>>> from api.models import Key
>>> Key.objects.filter(id = '1c5301e127f6')
[<Key: ssh-rsa ...>, <Key: ssh-rsa ...>, <Key: ssh-rsa ...>]
>>> Key.objects.filter(id = '1c5301e127f6').delete()
```
Where `1c5301e127f6` is my duplicate key id.

**Comment by bacongobbler** [comment]
This looks to have been fixed in v1.12.3 with https://github.com/deis/deis/pull/3530. Closing!
